src/scenario/scenario.c

```diff
@@ -7,36 +7,124 @@
 #include "game/save_version.h"
 #include "game/settings.h"
 #include "scenario/data.h"
+#include "scenario/request.h"
 
 struct scenario_t scenario;
 
+static struct {
+    int size;
+    int start_info;
+    int old_filler_1;
+    int invasions_part1;
+    int funds_enemy;
+    int map_size;
+    int briefing;
+    int old_filler_2;
+    int image;
+    int herds;
+    int demand_changes_part1;
+    int price_changes_part1;
+    int glademp;
+    int random_events;
+    int fishing;
+    int old_filler_3;
+    int invasions_part2;
+    int old_filler_4;
+    int rome_wheat;
+    int allowed_buildings;
+    int win_criteria;
+    int map_points;
+    int invasions_part3;
+    int misc;
+    int requests;
+    int custom_name;
+    int end;
+} state_offsets;
+
 int scenario_is_saved(void)
 {
     return scenario.is_saved;
 }
 
+static void state_offsets_init()
```

Comment:
```suggestion
static void state_offsets_init(void)
```

Reply:
I think you could use this function to get the offsets for every possible scenario version.

That would make it easier to manage even without using the `filler` variables you mentioned.

You could do something similar to the struct in `file_io.c`, which has a `features` struct, which would allow you to distinguish which versions have which stuff, and where they are located in the buffer.

Reply:
I'll change this if the removal of the filler bytes are needed.

Reply:
Done.
I didn't do a full features struct, but I updated the offset function to take version into account.